pkg/admission/mutator/webhook.go

```diff
@@ -59,7 +61,8 @@ func createCiliumPredicate() predicate.Funcs {
 			return false
 		}
 
-		return shoot.Spec.Networking.Type == cilium.ReleaseName
+		return !gardencorev1beta1helper.IsWorkerless(shoot) && shoot.Spec.Networking != nil &&
```

Comment:
Is this not a bit redundant? Why checking for whether the shoot is workerless when we check for the networking spec afterwards, which will most likely be `nil` for workerless shoots?
Is it just to highlight the case when the networking spec can be `nil` or did you have something else in mind?

Reply:
Adapted in 81fda70bd7b12a2716fb6fc53d9f869d3a82b034